docs/configuration.md

```diff
@@ -1,130 +1,129 @@
-via `S3_GW_WALLET_PASSPHRASE` variable or you will be asked to enter a password interactively. 
-You can also specify an account address to use from a wallet using `--address` parameter.
+Wallet (`--wallet`) is a mandatory parameter. It is a path to a wallet file. You can provide a passphrase to decrypt 
+a wallet via env variable or conf file, or you will be asked to enter a password interactively.
+You can also specify an account address to use from a wallet using the `--address` parameter.
 
-## Binding and TLS
+### Listening on address and TLS
 
-Gateway binds to `0.0.0.0:8080` by default, and you can change that with
-`--listen_address` option.
+Gateway listens on `0.0.0.0:8080` by default, and you can change that with the `--listen_address` option.
 
 It can also provide TLS interface for its users, just specify paths to the key and
 certificate files via `--tls.key_file` and `--tls.cert_file` parameters. Note
 that using these options makes gateway TLS-only, if you need to serve both TLS
 and plain text you either have to run two gateway instances or use some
 external redirecting solution.
 
-Example to bind to `192.168.130.130:443` and serve TLS there (keys and nodes
+Example to bind to `192.168.130.130:443` and serve TLS there (keys and nodes are
 omitted):
 
-```
+```shell
 $ neofs-s3-gw --listen_address 192.168.130.130:443 \
   --tls.key_file=key.pem --tls.cert_file=cert.pem
 ```
 
-## Monitoring and metrics
+### RPC endpoint and resolving of bucket names
 
-Pprof and Prometheus are integrated into the gateway, but not enabled by
-default. To enable them, use `--pprof` and `--metrics` flags or
-`S3_GW_PPROF`/`S3_GW_METRICS` environment variables.
+To set RPC endpoint specify a value of parameter `-r` or `--rpc-endpoint`. The parameter is **required if** another 
+parameter's `--resolve-order` value contains `nns`.
 
-## Yaml file
-Configuration file is optional and can be used instead of environment variables/other parameters. 
-It can be specified with `--config` parameter:
-```
-$ neofs-s3-gw --config your-config.yaml
+```shell
+$ neofs-s3-gw --rpc-endpoint http://morph-chain.neofs.devenv:30333/ --resolve-order nns,dns
 ```
 
-Configuration file example:
+### Processing of requests
+
+Maximum number of clients whose requests can be handled by the gateway can be specified by the value of 
+`--max_clients_count` parameter, the default value is 100. 
+`--max_clients_deadline` defines deadline after which the gate sends error `RequestTimeout` to a client, default value 
+is 30 seconds.
+
+```shell
+$ neofs-s3-gw --max_clients_count 150 --max_clients_deadline 1m
 ```
-listen_address: 0.0.0.0:8084
 
-wallet:
-  passphrase: 123456
+### Connection to NeoFS
 
-logger:
-  level: debug
+Timeout to connect to NeoFS nodes can be set with `--connect_timeout` (default 30s)
+and timeout to check node health during rebalance`--request_timeout` (default 15s).
```

Comment:
Maybe we should give more clear names to these parameters. For example, `dial_timeout` and `healthcheck_timeout`?

Reply:
I'm agree about `healthcheck_timeout` but I'm not sure that `dial_timeout` sounds clearer than `connect_timeout`

Reply:
I mentioned  `dial_timeout` since this name is used in SDK parameters https://github.com/nspcc-dev/neofs-s3-gw/blob/master/cmd/s3-gw/app.go#L108

Reply:
Yep, but I think we shouldn't take into account internal names in SDK, because a person, who starts the gateway, may not be in context of SDK. And for that user `connect_timeout` sounds more familiar than `dial_timeout`. 

Could you please join the discussion, @alexvanin? :)